libevmcore/Assembly.cpp

```diff
@@ -171,61 +173,96 @@ ostream& dev::eth::operator<<(ostream& _out, AssemblyItemsConstRef _i)
 	return _out;
 }
 
-ostream& Assembly::streamRLP(ostream& _out, string const& _prefix) const
+string Assembly::getLocationFromSources(StringMap const& _sourceCodes, SourceLocation const& _location) const
+{
+	if (_location.isEmpty() || _sourceCodes.empty())
+		return "";
+
+	string toReturn = _sourceCodes.find(*_location.sourceName)->second;
+
+	if (_location.end < (int)toReturn.size() - 1)
+			return "";
+
+	toReturn = toReturn.substr(_location.start, _location.end);
+	auto newLinePos = toReturn.find_first_of("\n");
+	if (newLinePos != string::npos && newLinePos != toReturn.size() - 1)
+		toReturn = toReturn.substr(0, newLinePos) + "...";
+
+	toReturn = "        " + toReturn;
+	return move(toReturn);
+}
+
+ostream& Assembly::streamRLP(ostream& _out, string const& _prefix, StringMap const& _sourceCodes) const
 {
-	_out << _prefix << ".code:" << endl;
+	auto appendLineWithPrefix = [&](string const& __sData, string const& _sourceCodeLine = string()) {
```

Comment:
Better not use a lambda here at all, this will also fix the compilation problems on Windows.